src/domain.c

```diff
@@ -679,23 +824,32 @@ enum OperationReturnCode ld_mod_entry_attrs(LDHandle *handle, const char *name,
     check_string(name, entry_name, "ld_mod_entry_attrs");
     check_string(parent, entry_parent, "ld_mod_entry_attrs");
 
-    TALLOC_CTX *talloc_ctx = talloc_new(NULL);
+    TALLOC_CTX *talloc_ctx = NULL; 
+    ld_talloc_new(talloc_ctx, error_exit, NULL);
 
     LDAPMod **attrs = fill_attributes(entry_attrs, talloc_ctx, opcode);
 
     const char* dn;
     if (strlen(prefix) > 0)
-    {
-        dn = talloc_asprintf(talloc_ctx,"%s=%s,%s", prefix, entry_name, entry_parent);
+    { 
+        ld_talloc_asprintf(dn, error_exit, talloc_ctx,"%s=%s,%s", prefix, entry_name, entry_parent);
     }
     else
     {
-        dn = talloc_asprintf(talloc_ctx,"%s,%s", entry_name, entry_parent);
+        ld_talloc_asprintf(dn, error_exit, talloc_ctx,"%s,%s", entry_name, entry_parent);
     }
 
     rc = modify(handle->connection_ctx, dn, attrs);
 
-    talloc_free(talloc_ctx);
+    ld_talloc_free(talloc_ctx, error_exit);
 
     return rc;
+
+    error_exit:
+        if (talloc_ctx)
+        {
+            ld_talloc_free(talloc_ctx, error_exit);
```

Comment:
Do not use error_exit mark in error_exit statement. For explanation Look for description of first similar issue at the beginning of the file.